apps/antalmanac/src/components/RightPane/CoursePane/SearchForm/DeptSearchBar/DeptSearchBar.tsx

```diff
@@ -1,175 +1,101 @@
-    };
-
-    handleSetDept = (event: ChangeEvent<unknown>, newDept: Department | null) => {
-        const setDeptValue = newDept === null ? options[0] : newDept;
-
-        this.setState({ value: setDeptValue });
-        RightPaneStore.updateFormValue('deptValue', setDeptValue.deptValue);
-        RightPaneStore.updateFormValue('deptLabel', setDeptValue.deptLabel);
-
-        const stateObj = { url: 'url' };
-        const url = new URL(window.location.href);
-        const urlParam = new URLSearchParams(url.search);
-        urlParam.delete('deptLabel');
-        urlParam.delete('deptValue');
-        if (
-            setDeptValue.deptValue &&
-            setDeptValue.deptValue != 'ALL' &&
-            setDeptValue.deptLabel &&
-            setDeptValue.deptLabel != 'ALL: Include All Departments'
-        ) {
-            urlParam.append('deptLabel', setDeptValue.deptLabel);
-            urlParam.append('deptValue', setDeptValue.deptValue);
-        }
-        const param = urlParam.toString();
-        const new_url = `${param.trim() ? '?' : ''}${param}`;
-        history.replaceState(stateObj, 'url', '/' + new_url);
-
-        if (newDept == null || newDept.deptValue === 'ALL') return;
-
-        const favorites = this.state.favorites;
-        let updatedFavorites = [...favorites];
-
-        if (favorites.filter((favorite) => newDept.deptValue === favorite.deptValue).length > 0) {
-            updatedFavorites.sort((a, b) => {
-                return a.deptValue === newDept.deptValue ? -1 : b.deptValue === newDept.deptValue ? 1 : 0;
-            });
-        } else {
-            updatedFavorites = [{ ...newDept, isFavorite: true }].concat(favorites);
-            if (updatedFavorites.length > 5) updatedFavorites.pop();
-        }
-        this.setState({ favorites: updatedFavorites });
-        setLocalStorageFavorites(JSON.stringify(updatedFavorites));
-    };
-
-    render() {
-        const { classes } = this.props;
-
-        return (
-            <div className={classes.formControl}>
-                <Autocomplete
-                    value={this.state.value}
-                    options={this.state.favorites.concat(options)}
-                    autoHighlight={true}
-                    openOnFocus={true}
-                    getOptionSelected={this.compareValues}
-                    getOptionLabel={(option) => option.deptLabel}
-                    onChange={this.handleSetDept}
-                    includeInputInList={true}
-                    noOptionsText="No departments match the search"
-                    groupBy={(dept) => (dept.isFavorite ? 'Recent Departments' : 'Departments')}
-                    renderInput={(params) => (
-                        <TextField {...params} label="Department" type="search" InputLabelProps={{ shrink: true }} />
-                    )}
-                />
-            </div>
-        );
-    }
-}
+            if (newValue == null || newValue === 'ALL') return;
```

Comment:
nit: redundant to check for null
```suggestion
            if (newValue === 'ALL') return;
```